src/components/app/App.js

```diff
@@ -18,130 +18,137 @@ import 'bootstrap/dist/css/bootstrap.min.css';
-
-      if (authorA.first_name > authorB.first_name) {
-        return 1;
-      }
-      if (authorA.first_name < authorB.first_name) {
-        return -1;
-      }
-      // a должно быть равным b
-      return 0;
-    });
-
-    this.setState({
-      posts: sorted
-    });
-  };
-
-  addPost = (newPost) => {
-    this.setState((prevState) => {
-      return {
-        posts: [{
-          ...newPost,
-          id: uniqueId(),
-        }, ...prevState.posts]
-      }
-    });
-  };
-
-  // todo 1: добавить здесь функцию onUserAdd
-  //  она должна добавлять пользователя в список users в стейте
-  //  при добавлении пользователя ему нужно добавить пропертю id, можно по аналогии со строкой 82
-
-
-  render() {
-    const { posts, selectedOption, users } = this.state;
-
-    return (
-      <div className="App">
-        <Header />
-
-        <PanelFromLecture label="Users" >
-          <AddUserForm/>
-          <UsersList users={users}/>
-        </PanelFromLecture>
-
-        <PanelFromLecture label="test" >
-          <PostPreview posts={posts} />
-        </PanelFromLecture>
-
-        <PanelFromLecture label="Posts">
-          <div className="d-flex">
-            Sorting:
-            <button onClick={this.onSortByAuthorClick}>By author</button>
-            <button onClick={this.onSortByDefault}>By default</button>
-
-            <DropDown
-              onSelect={this.onSort}
-              selectedOption={selectedOption}
-              options={sortingOptions}
-            />
-          </div>
-          <div className="d-flex posts-container">
-
-            <AddPostForm onAddPost={this.addPost} users={users} />
-
-            {
-              posts.map((item, index) => {
-                const user = usersList.find(user => user.id === item.user_id);
-                const author = user ? `${user.first_name} ${user.last_name}` : '';
-                const comments = allComments.filter(comment => comment.post_id === item.id);
-
-                return <Card
-                  post={item}
-                  key={item.id}
-                  author={author}
-                  comments={comments}
-                />;
-              })
+    };
+
+    addPost = (newPost) => {
+        this.setState((prevState) => {
+            return {
+                posts: [{
+                    ...newPost,
+                    id: uniqueId(),
+                }, ...prevState.posts]
             }
-          </div>
-        </PanelFromLecture>
-        <Footer />
-      </div>
-    );
-  }
+        });
+    };
+
+    // ttodo 1: добавить здесь функцию onUserAdd
+    //  она должна добавлять пользователя в список users в стейте
+    //  при добавлении пользователя ему нужно добавить пропертю id, можно по аналогии со строкой 82
+    onUserAdd = (newUser) => {
+        const users = this.state.users;
+        users.unshift({id:uniqueId(),...newUser}); // чи має право на життя цей вар-т?, чи не має мінусів?
```

Comment:
It's totally ok but you need to copy array of users in a line above so it would be `const users = [...this.state.users];`